client/src/app/core/services/http.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+export enum HTTPMethod {
+    PUT,
+    PATCH
+}
+
+/**
+ * Send data back to server
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class HttpService {
+    /**
+     * Construct a DataSendService
+     *
+     * @param http The HTTP Client
+     */
+    public constructor(private http: HttpClient) {}
+
+    public create(url: string, obj: object): Observable<object> {
+        url = this.formatForSlash(url);
+        return this.http.post<object>(url, obj).pipe(
+            tap(
+                response => {
+                    // TODO: Message, Notify, Etc
+                    console.log('New Model added. Response :\n ', response);
+                },
+                error => console.log('Error:\n ', error)
+            )
+        );
+    }
+
+    private formatForSlash(str: string): string {
```

Comment:
comment.

also:
`return str.endsWith('/') ? str : str+='/';`

(or rather don't use at all, since you manipulate the original string aswell)